openfisca_aotearoa/variables/acts/social_security/disability_allowance/disability_allowance.py

```diff
@@ -0,0 +1,353 @@
+
+
+class disability_allowance__income_limit_clause_14(variables.Variable):
+    value_type = bool
+    default_value = False
+    entity = entities.Person
+    definition_period = periods.WEEK
+    label = "Does the person earn below the Disability Allowance Income Limit as per clause 14?"
+    reference = "https://www.legislation.govt.nz/act/public/2018/0032/latest/DLM6784890.html"
+
+    def formula_2018_11_26(persons, period, parameters):
+        sole_parent = persons("is_sole_parent", period) * \
+        persons("disability_allowance__sole_parent_meets_relationship_qualification", period)
+        more_than_one_child = (persons("social_security__dependent_children", period) > 1)
+        income_within_limit = persons("social_security__income", period) <= parameters(period).disability_allowance.income_limits.single_person.sole_parent_with_one_dep_child
+        return more_than_one_child * sole_parent * income_within_limit
+
+
+class disability_allowance__is_adequately_supported_by_partner(variables.Variable): #copied, need to refactor 
+    value_type = bool
+    entity = entities.Person
+    default_value: False
+    label = """Is adequately supported by their partner? (false if lost the regular support of
+    their partner as their partner has been imprisoned or is subject to release or detention conditions that prevent employment)
+    """
+    definition_period = periods.WEEK
+    reference = "https://www.workandincome.govt.nz/map/income-support/main-benefits/sole-parent-support/qualifications.html"
+
+
+class disability_allowance__sole_parent_meets_relationship_qualification(variables.Variable): #copied, need to refactor 
+    value_type = bool
+    default_value = True
+    entity = entities.Person
+    label = "Meets the sole parent support test for not being in a relationship"
+    definition_period = periods.WEEK
+    reference = "https://www.workandincome.govt.nz/map/income-support/main-benefits/sole-parent-support/qualifications.html"
+    """
+     be one of the following:
+        * living apart from their partner and lost the support or being inadequately
+            maintained by the spouse or partner
+        * divorced or had their civil union dissolved
+        * single (never had a partner)
+        * has lost the regular support of their partner as their partner has been imprisoned or
+            is subject to release or detention conditions that prevent employment or
+        * their spouse or partner has died
+    """
+    def formula(persons, period, parameters):
+        # Do they have a partner
+        no_partners = (persons("disability_allowance__person_has_partner", period) == 0)
+        not_supported = (persons("disability_allowance__is_adequately_supported_by_partner", period) == 0)
+        # no partner, OR not supported by partner
+        return no_partners + not_supported
+
+
+class disability_allowance__current_income(variables.Variable):
```

Comment:
Check if social_security__income is sufficient

Reply:
already used that one! this one was an extra variable and has now been removed.